### ISSUE TITLE
custom vault URL

### ISSUE THREAD

**Issue by WouterLS** [anonymous1184/bitwarden-autotype#4]
Hi,

can this be used on a self-hosted vault with a custom URL?

**Comment by anonymous1184** [comment]
Yes, it works the same regarding of the login server. However I didn't tough of that so currently there's no option within the application.

I'll update to add this as is just a few lines of code, in the meantime you can issue the following commands:

```bat
cd "
"
"
';'"
"
%AppData%
```

**Comment by anonymous1184** [comment]
Yes, it works the same regarding of the login server. However I didn't tough of that so currently there's no option within the application.

I'll update to add this as is just a few lines of code, in the meantime (once Auto-Type is installed), you can issue the following commands in a terminal:

```cmd
cd /d "%AppData%\Auto-Type"
set "BITWARDENCLI_APPDATA_DIR=%CD%"
bw logout
bw config server
bw config server https://your.self.hosted.server:1234
bw config server
```

The output should be something like this:

![20210602104342](https://user-images.githubusercontent.com/53758552/120510843-6e1bfe00-c38f-11eb-80bc-99297a450687.png)


Indicating the default server and then your server. After that you should be able to login with both the CLI and Auto-Type.

**Comment by Nanakie** [comment]
It works for me via the cli when I check it
but auto-type always comes up with the wrong username or password

**Comment by myrandor** [comment]
+1

Same here. I made the changes with the CLI. The data.json file get changed correctly.

However, when I start Auto-Type, it create a new data.json file even if a file was already there...

**Comment by myrandor** [comment]
Oh! Just find a fix!

After doing the "bw config server" command, you have to login using CLI and "SYNC" in order to populate the JSON correctly.

`cd /d "%AppData%\Auto-Type"
set "BITWARDENCLI_APPDATA_DIR=%CD%"
bw logout
bw config server
bw config server https://your.self.hosted.server:1234
bw config server
bw login <email_address> <password>
bw sync --session <session_id provided using the last command>`

After that, Auto-Type was able to login the my personal server.

**Comment by anonymous1184** [comment]
The next update will include this, I uploaded the code almost a month ago https://github.com/anonymous1184/bitwarden-autotype/commit/a22cbdddd94d24c350f8291e5499aa9cc8eda435.

But I had a issue I haven't been able to reproduce, so I'm holding it... however if I cannot reproduce it again I'll will do the release.

Thanks a lot for the info on the manual fix, and as always since I started with the internet almost 30 years ago... Canadians are just awesome.

**Comment by myrandor** [comment]
Just had a strange issue. It seems like after a reboot or update, I had to redo the procedure again... The "server" configuration was not kept... :(

**Comment by KnightTim** [comment]
to add one note to this issue, if you have a custom vault and click the "Open vault" right click menu option it will take you to the standard bitwarden vault page.

**Comment by anonymous1184** [comment]
@KnightTim indeed, that an omission on my end. The next batch of updates is already on queue and will include this. Thanks.

**Comment by Nypheena** [comment]
> @KnightTim indeed, that an omission on my end. The next batch of updates is already on queue and will include this. Thanks.

hows it coming along for the update?
